src/main/python/tests/nn_nodes/test_affine.py

```diff
@@ -0,0 +1,67 @@
+# -------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# -------------------------------------------------------------
+
+import unittest
+
+import numpy as np
+
+from systemds.context import SystemDSContext
+
+from systemds.operator.nn_nodes.affine import Affine
+
+dim = 6
+n = 10
+m = 5
+np.random.seed(11)
+X = np.random.rand(n, dim)
+
+np.random.seed(10)
+W = np.random.rand(dim, m)
+b = np.random.rand(m)
+
+
+class TestAffine(unittest.TestCase):
+    sds: SystemDSContext = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.sds = SystemDSContext()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.sds.close()
+
+    def test_affine(self):
```

Comment:
add a dedicated tests for forward for backwards.
Ideally you add multiple such tests.

Then write in what you expect to get out from the operation based on a explicit input.